### PR TITLE
Fixed #32580 -- Doc'd that HttpRequest.get_host() may raise DisallowedHost.

### PR DESCRIPTION
**Pull request by sreehari1997**
- Documented HttpRequest.get_host() can raise DisallowedHost exception.
- Add contributor name to AUTHORS.

Ticket [#32580](https://code.djangoproject.com/ticket/32580)

### CODE REVIEW

**Comment by sreehari1997** [comment]
Rendered documentation for ```get_host()```
Screenshot 2021-03-25 at 2 08 09 PM

File: docs/ref/request-response.txt [hunk 1]
```diff
@@ -290,6 +290,9 @@ Methods
 
     Example: ``"127.0.0.1:8000"``
 
+    Raises a ``django.core.exceptions.DisallowedHost`` exception if the host
+    is invalid.
```
Comment (cjerdonek) [inline docs/ref/request-response.txt:294]: I would explicitly mention "... and not in [`ALLOWED_HOSTS`](https://docs.djangoproject.com/en/3.1/ref/settings/#allowed-hosts)" so people understand this isn't just about whether it parses correctly. See here for the code / error messages that `get_host()` provides: https://github.com/django/django/blob/41e6b2a3c5e723256506b9ff49437d52a1f3bf43/django/http/request.py#L130-L135

Comment (sreehari1997) [reply docs/ref/request-response.txt]: @cjerdonek I've made the changes.

**Comment by sreehari1997** [comment]
Rendered documentation for ```get_host()```
Screenshot 2021-03-25 at 3 14 29 PM

File: docs/ref/request-response.txt [hunk 2]
```diff
@@ -290,6 +290,10 @@ Methods
 
     Example: ``"127.0.0.1:8000"``
 
+    Raises a ``django.core.exceptions.DisallowedHost`` exception if the host
+    is not in ``ALLOWED_HOSTS`` or the domain name is not valid according to
+    ``RFC 1034/1035``.
```
Comment (cjerdonek) [inline docs/ref/request-response.txt:295]: I think `DisallowedHost` and `ALLOWED_HOSTS ` should be properly linked to their entry in the docs. You might as well also make the two RFC numbers external links, if you're including those in the text.

Comment (sreehari1997) [reply docs/ref/request-response.txt:295]: @cjerdonek can I use the following links for RFC 1034 and 1035.
- https://tools.ietf.org/html/rfc1034
- https://tools.ietf.org/html/rfc1035

Comment (cjerdonek) [reply docs/ref/request-response.txt:295]: Sure, I think so.

**Comment by felixxm** [comment]
@sreehari1997 Thanks for this patch :+1: Welcome aboard :boat: 

I added references.

**Pull request merged**